### PR TITLE
Add user to project endpoint.

### PR DESCRIPTION
**Pull request by Rishanthakumar**
This will add an endpoint to assign users to project.

### CODE REVIEW

File: src/api/projects.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ import {done} from '../helpers/response-handler';
  * @param  {object} event
  * @param  {object} context
  * @param  {object} callback
- * @return {function} done
+ * @return {function} callback
```
Comment (singhabahu) [inline src/api/projects.js]: It should be response for the API proxy

File: serverless.yml [hunk 2]
```diff
@@ -38,6 +38,14 @@ functions:
           method: POST
           authorizer: auth
           cors: true
+  add-project-user:
+    handler: 'src/api/projects.addUser'
```
Comment (singhabahu) [inline serverless.yml]: can't have insert here rather than addUser?

File: src/api/projects.js [hunk 3]
```diff
@@ -67,7 +67,95 @@ export const create = (event, context, callback) => {
...
+
+       User.find({
+        attributes: ['uuid', 'organizationId'],
+        where: {
+          uuid: userId,
+        },
+      }).then((user) => {
+        Project.find({
+          where: {
+            id: projectId,
```
Comment (singhabahu) [inline src/api/projects.js:114]: can't we do a where clause to match both projectId and the organizationId?

File: src/api/projects.js [hunk 4]
```diff
@@ -67,7 +67,95 @@ export const create = (event, context, callback) => {
...
+          },
+        }).then((project) => {
+            if (project.organizationId == user.organizationId) {
+              project.addUser(user).then((result) => {
+                if (result.length == 0) {
+                  return callback(null, done({
+                    statusCode: 400,
+                    message: `User has already added to this project`,
```
Comment (singhabahu) [inline src/api/projects.js]: change to `User has been already added to this project`

File: src/api/projects.js [hunk 5]
```diff
@@ -67,7 +67,95 @@ export const create = (event, context, callback) => {
...
+              }).catch((error) => {
+                return callback(null, done({
+                  statusCode: 500,
+                  message: error,
+                }));
+              });
+            } else {
+              return callback(null, done({
+                statusCode: 400,
+                message: `User and project not belong to the same organization`,
```
Comment (singhabahu) [inline src/api/projects.js]: Don't do this, people can do request with random ids and get project details. Just say bad request or project not found. Fixing the above mentioned where clause issue will resolve this one as well

File: src/util/permission.js [hunk 6]
```diff
@@ -21,11 +21,15 @@ const hasPermission = (userId, permission) => {
...
             }],
         }).then((result) => {
-            resolve(
-                JSON.parse(
-                    result.role.permission
-                )[permission.realm].includes(permission.action)
-            );
+            if (result == null) {
+                reject('Cannot find role for the given id');
+            } else {
```
Comment (singhabahu) [inline src/util/permission.js:26]: Cant we simplify this into one line like 
```JS
if (result == null) return reject('Cannot find role for the given id');
resolve()
```

**Review by singhabahu — CHANGES_REQUESTED** [review]

File: src/util/permission.js (hunk 6 above)
Comment (Rishanthakumar) [reply src/util/permission.js:26]: @singhabahu if i do this it shows a maximum length lint issue. So i think better to keep as it is.

**Review by singhabahu — APPROVED** [review]
Looks good to me. We need to make these queries as transactions pretty soon

**Pull request merged**